### ISSUE TITLE
Compile time `MMark`

### ISSUE THREAD

**Issue by srid** [mmark-md/mmark#55]
I'd like to embed my markdown content in the Haskell source in a manner similar to `file-embed` but (obviously) as `MMark` documents instead of `ByteString`.

The primary benefit here is that we fail **at compile time** if any of the markdown files (in the repository) fails to parse.

Would implementing this feature in this primary repo be within its scope, or is it best done as a separate library? I'm only beginning to get familiar TH, so not yet sure how trivial to implement this would be.

**Comment by mrkkrp** [comment]
Yes, I guess this is in scope. I thought of something like this myself, but decided that would be not a very popular use case, but you seem to confirm that it's not so. For inspiration see:

* [Template Haskell tutorial](https://markkarpov.com/tutorial/th.html)
* Stache, which can parse Mustache templates at compile time: https://hackage.haskell.org/package/stache

Hope that helps.

**Comment by mrkkrp** [comment]
What happened to this use case? Have you discovered that you don't need this or have you implemented this functionality but never opened a PR?

**Comment by srid** [comment]
@mrkkrp I never implemented it.
